test/e2e/e2e_test.go

```diff
@@ -79,6 +85,9 @@ var _ = BeforeSuite(func() {
 		// This flag can be deprecated in favor of the kubeconfig provisioner:
 		os.Setenv("KUBECONFIG", *kubeConfigPath)
 	}
+	if collectArtifactsScriptPath != nil && *collectArtifactsScriptPath != "" {
+		os.Setenv("E2E_ARTIFACT_SCRIPT", *collectArtifactsScriptPath)
+	}
 
```

Comment:
Note: just a hack to inject state so the ctx package can derive where to find the artifacts bash script path location.